src/components/Header.tsx

```diff
@@ -0,0 +1,52 @@
+import { useEffect } from 'react';
+
+interface Props {
+  updateAll: () => void;
+  handleAdd: () => void;
+  editTodo: string;
+  setEditTodo: (a: string) => void;
+  disabledInput: boolean;
+  inputRef: React.RefObject<HTMLInputElement>;
+}
+
+export const Header: React.FC<Props> = ({
+  updateAll,
+  handleAdd,
+  editTodo,
+  setEditTodo,
+  disabledInput,
+  inputRef,
+}) => {
+  useEffect(() => {
+    inputRef.current?.focus();
+  }, []);
+
+  return (
+    <header className="todoapp__header">
+      <button
+        type="button"
+        className="todoapp__toggle-all"
+        data-cy="ToggleAllButton"
+        onClick={updateAll}
+      ></button>
+
+      <form
+        onSubmit={e => {
+          e.preventDefault();
+          handleAdd();
+        }}
```

Comment:
```suggestion
        onSubmit={handleAdd}
```

you can make handleAdd as function which except event.

```
  const handleAdd = (e: FormEvent) => {
     e.preventDefault();
     // do something
  }
```